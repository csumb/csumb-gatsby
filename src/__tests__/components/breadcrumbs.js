import React from 'react'
import renderer from 'react-test-renderer'
import Breadcrumbs from 'components/header/breadcrumbs'

const sampleBreadcrumbs = JSON.stringify([
  {
    href: '/',
    title: 'CSUMB Home',
  },
  {
    title: 'About CSUMB',
    href: '/about',
  },
])

describe('Breadcrumbs', () => {
  it('renders breadcrumbs correctly', () => {
    const tree = renderer
      .create(<Breadcrumbs breadcrumbs={sampleBreadcrumbs} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('returns null for no breadcrumbs', () => {
    const tree = renderer.create(<Breadcrumbs breadcrumbs={''} />).toJSON()
    expect(tree).toBeNull()
  })
})
