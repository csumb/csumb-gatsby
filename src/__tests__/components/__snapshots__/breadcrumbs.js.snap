// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs renders breadcrumbs correctly 1`] = `
<ul
  className="css-1sct6h1"
>
  <li
    className="css-e3hzvr"
  >
    <a
      aria-current="page"
      className=""
      href="/"
      onClick={[Function]}
      onMouseEnter={[Function]}
      style={Object {}}
    >
      CSUMB Home
    </a>
  </li>
  <li
    className="css-e3hzvr"
  >
    <a
      href="/about"
      onClick={[Function]}
      onMouseEnter={[Function]}
    >
      About CSUMB
    </a>
  </li>
</ul>
`;
